### PR TITLE
BusBud Challenge Submission

### PR DESCRIPTION
**Pull request by Rolvar**
Hi BusBuddies, 

Here is my submission for the challenge, I hope you guys find it well, I explain my solution in the README file and you can pull this and run as is (all its dependencies are in the project).

Crossing Fingers!! 

Best Regards,

### CODE REVIEW

File: busbudchallenge/busbudchallenge/support files/SceneDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    var window: UIWindow?
+
+    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+        guard let windowScene = (scene as? UIWindowScene) else { return }
+        window = UIWindow(windowScene: windowScene)
+        window?.rootViewController = RootViewController()
+        window?.makeKeyAndVisible()
+        if #available(iOS 13.0, *) {
+            window?.overrideUserInterfaceStyle = .light
+        }
```
Comment (xzoky) [inline busbudchallenge/busbudchallenge/support files/SceneDelegate.swift:21]: Excuse me _WHAT_? 😨

File: busbudchallenge/busbudchallenge/views/results/subviews/ResultsTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    func setupContent() {
+        guard let item = item else { return }
+        operatorLogo.downloaded(from: item.operatorLogo)
+        operatorName.text = item.operatorName
+        departureTime.text = item.departureTime
+        departureLocationName.text = item.departureLocation
+        arrivalTime.text = item.arrivalTime
+        arrivalLocationName.text = item.arrivalLocation
+        currencySymbol.text = item.currencySymbol
+        price.text = item.ticketPrice
```
Comment (xzoky) [inline busbudchallenge/busbudchallenge/views/results/subviews/ResultsTableViewCell.swift:175]: No!

File: README.md [hunk 3]
```diff
@@ -2,258 +2,61 @@
...
 
-- `origin` : Origin's geohash
-- `destination` : Destination's geohash
-- `outbound_date` : Outbound departure date, ISO 8061 date
+Even tho I'm not a designer, I tried to apply as much effort as possible to the UI/UX, because a very good looking app helps to retain users, also its behavior. So I added some UI tweaks in the code, for example the `UIImage.blur()` function and so many others so can help to build a good looking app.
 
-QUERY PARAMS:
+## Third Party Libraries
+Although some companies doesn't like to use third-party libraries inside of their projects, there are some libraries that are needed to not *reinvent the wheel*. In this project I just use 4 libraries:
+* **Alamofire** to handle the API requests
```
Comment (xzoky) [inline README.md:47]: Probably unnecessary, you're only using it to create a single GET request, that's like 2 lines!

File: README.md [hunk 4]
```diff
@@ -2,258 +2,61 @@
...
+* **SnapKit** to create UI constraints programmatically and very easy
+* **Hero** This library is used for the transitions, but I would rather not to use it, is under development, but offers a good set of transitions that are very easy to implement.
```
Comment (xzoky) [inline README.md:49]: SwiftUI is "too risky", but this is A-OK??

**Review by xzoky — COMMENTED** [review]
## TL;DR

No polling at all, inscrutable view layout code, uninspired design, questionable technical choices. Nothing impressive to redeem the bad stuff.
This is not good enough for a senior position.

---

## Good
* Localized strings for 3 languages.

## Bad
* No polling whatsoever. The candidate did not follow the instructions.
* Committed the API token.
* Vast majority of API model properties are optional when they don't need to be.
* Incorrect date / price formatting.
* Messy typography, should have stuck with system defaults.
* Zero tests.

## Ugly
* Typos.
* So much view layout code, so much repetition, lots of it inside controller implementations.
* App is manually forced out of dark mode.
